pkg/util/resources.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	//"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+
+	"github.com/kubestellar/kubestellar/api/control/v1alpha1"
+)
+
+// ParseAPIGroupsString takes a comma separated string list of api groups in the form of
+// <api-group1>, <api-group2> .. and returns a sets.Set[string]
+func ParseAPIGroupsString(apiGroups string) sets.Set[string] {
+	if apiGroups == "" {
+		return nil
+	}
+
+	groupsSet := sets.Set[string]{}
+	for _, g := range strings.Split(apiGroups, ",") {
```

Comment:
FYI: https://github.com/kubernetes/apimachinery/blob/v0.28.6/pkg/util/sets/set.go#L31 makes the loop here unnecessary.

Reply:
You are saying to do something like `sets.New(strings.Split(apiGroups, ","))` ? But that returns sets.Set[[]string], which I am not sure how to use

Reply:
`sets.New(strings.Split(apiGroups, ",")...)`

Reply:
Ok, thanks, updated